Jenkinsfile

```diff
@@ -5,7 +5,27 @@ pipeline {
         parallelsAlwaysFailFast()
     }
     stages {
-
+        stage('Ready For CI') {
+            agent {
+                docker {
+                    image 'terrier:focal'
+                }
+            }
+            when {
+                not {
+                    branch 'master'
+                }
+            }
+            steps {
+                script {
+                   ready_for_build = sh script: 'python3 ./build-support/check_github_labels.py', returnStatus: true
+                   if(ready_for_build != 0) {
+                        currentBuild.result = 'ABORTED'
+                        abort('Not ready for CI. Please add ready-for-ci tag in Github when you are ready to build your PR.')
+                   }
+                }
+            }
```

Comment:
I think we may want to cleanup the directory:

```
                    post {
                        cleanup {
                            deleteDir()
                        }
                    }
```

Reply:
I added it. I'm still slightly concerned about the random agent bottleneck that I mentioned. Not sure what a good fix is though.

Reply:
I'm not particularly concerned with that agent bottleneck; we have a lot of those terrier:focal's lying around. Unless you have opinions, I would be OK with merging as is.

Reply:
Yeah, sorry didn't see this comment. After talking with Dan I think this is fine.